Algorithm/QCAlgorithm.Indicators.cs

```diff
@@ -1268,44 +1269,9 @@ public WilderMovingAverage WWMA(Symbol symbol, int period, Resolution? resolutio
         /// <param name="type">The indicator type, for example, 'SMA5'</param>
         /// <param name="resolution">The resolution requested</param>
         /// <returns>A unique for the given parameters</returns>
-        public string CreateIndicatorName(Symbol symbol, string type, Resolution? resolution)
+        public string CreateIndicatorName(Symbol symbol, string type, ResolutionTimeSpan resolution)
         {
-            if (!resolution.HasValue)
-            {
-                resolution = GetSubscription(symbol).Resolution;
-            }
-            string res;
-            switch (resolution)
-            {
-                case Resolution.Tick:
-                    res = "_tick";
-                    break;
-
-                case Resolution.Second:
-                    res = "_sec";
-                    break;
-
-                case Resolution.Minute:
-                    res = "_min";
-                    break;
-
-                case Resolution.Hour:
-                    res = "_hr";
-                    break;
-
-                case Resolution.Daily:
-                    res = "_day";
-                    break;
-
-                case null:
-                    res = string.Empty;
-                    break;
-
-                default:
-                    throw new ArgumentOutOfRangeException("resolution");
-            }
-
-            return string.Format("{0}({1}{2})", type, symbol.ToString(), res);
+            return string.Format("{0}({1}{2})", type, symbol.ToString(), resolution.ToString());
```

Comment:
You should keep the previous format ("_tick", "_sec", ..., "_day"). 
If it is not one of those cases, then we could use `.ToString()`.
You can add an unit test for this method.

Reply:
Hmm but it will never be in the Resolution format when it reaches this function. The ResolutionTimespan class only converts to TimeSpan.
So it will always see TimeSpan and it won't match any of the cases.

Reply:
Why not something like this?
```csharp
if (resolution == TimeSpan.Zero)
{
   return $"{type}{symbol}_tick";
}
else if (resolution == QuantConnect.Time.OneSecond)
{
    return $"{type}{symbol}_sec";
}
else if (resolution == QuantConnect.Time.OneMinute)
{
    return $"{type}{symbol}_min";
}
// other cases
else
{
    return $"{type}{symbol}{resolution}";
}
```

Reply:
There's a function for this :)  Checkout `Extensions.ToHigherResolutionEquivalent(false)` -- might be able to use the output of that, and then determine how many 'units' of the resolution fit into the timespan, so if you have 5 minutes, then `ToHigherResolutionEquivalent(false)` will return Resolution.Minute, and then 5minutes/Resolution.Minute.ToTimeSpan() => 5.

Anyway, this is certainly not incredibly important here

Reply:
I still like my solution the most because there is no simple way to take care of 1:30:00.
What should 1 hour and 30 mins return?